### PR TITLE
MODLD-215: Use both $4 and $e/$j subfields for creator/contributor re…

### PR DESCRIPTION
**Pull request by askhat-abishev**
…lations

### CODE REVIEW

File: src/test/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImplTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+      @Override
+      public Optional<Character> getText() {
+        return Optional.of('e');
+      }
+    };
+  }
+
+  private static Subfield getSubfield(String data) {
+    return new Subfield() {
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/test/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImplTest.java]: done

File: src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java [hunk 2]
```diff
@@ -28,15 +30,18 @@ public void checkRelation(Resource source,
...
+        .forEach(predicate -> source.getOutgoingEdges().add(new ResourceEdge(source, target, predicate)))
+      );
   }
 
-  private Optional<String> getPredicate(Relation relation, DataField dataField) {
-    return getByCode(relation, dataField)
-      .or(() -> getByText(relation, dataField));
+  private List<String> getPredicates(Relation relation, DataField dataField) {
+    return Stream.of(getByCode(relation, dataField), getByText(relation, dataField))
+      .filter(Optional::isPresent)
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java]: done

File: src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java [hunk 3]
```diff
@@ -28,15 +30,18 @@ public void checkRelation(Resource source,
...
+        .stream()
+        .map(PredicateDictionary::valueOf)
+        .forEach(predicate -> source.getOutgoingEdges().add(new ResourceEdge(source, target, predicate)))
+      );
   }
 
-  private Optional<String> getPredicate(Relation relation, DataField dataField) {
-    return getByCode(relation, dataField)
-      .or(() -> getByText(relation, dataField));
+  private List<String> getPredicates(Relation relation, DataField dataField) {
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java:41]: done

File: src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java [hunk 4]
```diff
@@ -28,15 +30,18 @@ public void checkRelation(Resource source,
...
                             Marc2ldFieldRuleApplier fieldRule) {
     fieldRule.getRelation()
-      .flatMap(relation -> getPredicate(relation, dataField))
-      .map(PredicateDictionary::valueOf)
-      .map(predicate -> new ResourceEdge(source, target, predicate))
-      .ifPresent(resourceEdge -> source.getOutgoingEdges().add(resourceEdge));
+      .ifPresent(relation -> getPredicates(relation, dataField)
+        .stream()
+        .map(PredicateDictionary::valueOf)
+        .forEach(predicate -> source.getOutgoingEdges().add(new ResourceEdge(source, target, predicate)))
```
[the remark this answers is not in the document]
Comment (askhat-abishev) [reply src/main/java/org/folio/marc4ld/service/marc2ld/relation/RelationProviderImpl.java]: done
https://github.com/FOLIO-EIS/lib-linked-data-dictionary/pull/60

**Review by aleksei-pronichev — APPROVED** [review]

**Review by PBobylev — APPROVED** [review]

**Pull request merged**